### ISSUE TITLE
TPS is completly random

### ISSUE THREAD

**Issue by Turmfalke2** [wormzjl/Project-PFAA#5]
hi,
I am playing in single player and I am still in the stone age, but I found the TPS to be completly random.
For example /tps just showed me this:
`
Average TPS over last 200 ticks: 15.17 (lowest: 0.30)  
Average TPS over last 20 ticks: 240.96 (lowest: 200.00)  
Average TPS over last 5 ticks: 250.00 (lowest: 250.00)  

`

**Comment by wormzjl** [comment]
What's you rig?
Also you can use opis to see the tps variations

**Comment by Turmfalke2** [comment]
Nothing special:
i5-6600K
gtx1070
32G ddr4

What is confusing me the most is when tps goes beyond 20. In game this looks like monsters just teleporting on top of me and hitting me 20 times with in one second. I have had cases were I have seen a bear several chunks away and in the next moment I was dead.

**Comment by wormzjl** [comment]
Seems like a lagspike to me, it's weired since I haven't seen anything like this during my test.
Can you try to add this to java parameters (in launcher config - java/minecraft)?
-XX:+UseG1GC

Also make sure you give java 4g memory, but no more than 6g.

**Comment by wormzjl** [comment]
http://files.player.to/sampler-1.60.1.jar
Can you drop this into the mods folder and play for a while.
This will generate a log if lagspike is detected, which is located in /stall-reports

**Comment by Turmfalke2** [comment]
Turns out I was already using the G1GC.
[stall-reports.zip](https://github.com/wormzjl/Project-PFAA/files/1528559/stall-reports.zip)

I felt like I had less lag issues today, while I still had some unexpected tps drops I never caught it above 20tps.

**Comment by wormzjl** [comment]
do you happens to be exploring the world everytime the lagspike hits?

**Comment by Turmfalke2** [comment]
Not really. I mean there is  a lot of exploring involved in playing the pack, but so far I haven't had issues with chunks not loading in time which is usually a pretty good indication that one exploring to fast.

**Comment by wormzjl** [comment]
just updated, check if it still lags with 1.0.2

**Comment by Turmfalke2** [comment]
Can't find a changelog, but you removed thaumcraft?! I am now going to have holes everywhere in my world?

**Comment by wormzjl** [comment]
Not really, just some floating cobwebs
Edit: back up your world before updating, just in case

**Comment by Turmfalke2** [comment]
Given that most of my house/almost all of my support beams are made of great wood I have elected not to update.

**Comment by wormzjl** [comment]
you can replace them by jumping into creative, people reported that this update really helps with these lagspikes (and crashes) and your fps
